Add TCPS support for Oracle DB (#10591)
DataDog/integrations-core: oracle/README.md

    only_custom_queries: true
```

#### Connecting to Oracle via TCPS

> To connect to Oracle via TCPS (TCP with SSL), make sure to first uncomment out the `protocol` configuration option and select `TCPS`, and make sure that the correct `server` is  serving the TCPS server.

Reviewer: ```suggestion
To connect to Oracle via TCPS (TCP with SSL), uncomment out the `protocol` configuration option and select `TCPS`.
Update the `server` option to set the TCPS server to monitor.
```

```yaml
init_config:

instances:
  ## @param server - string - required
  ## The IP address or hostname of the Oracle Database Server.
  #
  - server: localhost:1522

    ## @param service_name - string - required
    ## The Oracle Database service name. To view the services available on your server,
    ## run the following query:
    ## `SELECT value FROM v$parameter WHERE name='service_names'`
    #
    service_name: "<SERVICE_NAME>"

    ## @param user - string - required
    ## The username for the user account.
    #
    user: <USER>

    ## @param password - string - required
    ## The password for the user account.
    #
    password: "<PASSWORD>"

    ## @param protocol - string - optional - default: TCP
    ## The protocol to connect to the Oracle Database Server. Valid protocols include TCP and TCPS.
    ##
    ## When connecting to Oracle Database via JDBC, `jdbc_truststore` and `jdbc_truststore_type` are required.
    ## More information can be found from Oracle Database's whitepaper:
    ##
    ## https://www.oracle.com/technetwork/topics/wp-oracle-jdbc-thin-ssl-130128.pdf
    #
    protocol: TCPS
```

> After uncommenting out the `protocol` configuration option, make sure that the `sqlnet.ora`, `listener.ora`, and `tnsnames.ora` are correctly configured to allow TCPS connections on your Oracle Database. 

Reviewer: Maybe move this up to the above paragraph as a continuation?

```suggestion
Update the `sqlnet.ora`, `listener.ora`, and `tnsnames.ora` to allow TCPS connections on your Oracle Database. 
```

> If you are connecting to Oracle Database using the Oracle Instant Client, you can verify that Datadog will be able to connect to your database by using the `sqlplus` command line tool using the information inputted in your configuration options:

Reviewer: ```suggestion
##### TCPS via Oracle Instant Client

If you are connecting to Oracle Database using the Oracle Instant Client, verify that the Datadog Agent will be able to connect to your database by using the `sqlplus` command line tool using the information inputted in your configuration options:
```

add subtitle to easier navigation and linking

```shell
sqlplus <USER>/<PASSWORD>@(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCPS)(HOST=<HOST>)(PORT=<PORT>))(SERVICE_NAME=<SERVICE_NAME>)))
```

> If you are connecting to Oracle Database using JDBC, you will need to also specify `jdbc_truststore_path`, `jdbc_truststore_type`, and optionally `jdbc_truststore_password` if there is a password on the truststore. Note that `SSO` truststores don't require passwords.
> 

Reviewer: ```suggestion
##### TCPS via JDBC

If you are connecting to Oracle Database using JDBC, you will need to also specify `jdbc_truststore_path`, `jdbc_truststore_type`, and optionally `jdbc_truststore_password` if there is a password on the truststore. 

Note: `SSO` truststores don't require passwords.

```

> ```yaml
>     ...

Reviewer: ```suggestion
```yaml
    # In the `instances:` section
    ...
```

    only_custom_queries: true
```

#### Connecting to Oracle via TCPS

> To connect to Oracle via TCPS (TCP with SSL), uncomment out the `protocol` configuration option and select `TCPS`.

Reviewer: ```suggestion
To connect to Oracle via TCPS (TCP with SSL), uncomment the `protocol` configuration option and select `TCPS`.
```

Update the `server` option to set the TCPS server to monitor.

```yaml
init_config:

instances:
  ## @param server - string - required
  ## The IP address or hostname of the Oracle Database Server.
  #
  - server: localhost:1522

    ## @param service_name - string - required
    ## The Oracle Database service name. To view the services available on your server,
    ## run the following query:
    ## `SELECT value FROM v$parameter WHERE name='service_names'`
    #
    service_name: "<SERVICE_NAME>"

    ## @param user - string - required
    ## The username for the user account.
    #
    user: <USER>

    ## @param password - string - required
    ## The password for the user account.
    #
    password: "<PASSWORD>"

    ## @param protocol - string - optional - default: TCP
    ## The protocol to connect to the Oracle Database Server. Valid protocols include TCP and TCPS.
    ##
    ## When connecting to Oracle Database via JDBC, `jdbc_truststore` and `jdbc_truststore_type` are required.
    ## More information can be found from Oracle Database's whitepaper:
    ##
    ## https://www.oracle.com/technetwork/topics/wp-oracle-jdbc-thin-ssl-130128.pdf
    #
    protocol: TCPS
```

> Update the `sqlnet.ora`, `listener.ora`, and `tnsnames.ora` to allow TCPS connections on your Oracle Database. 

Reviewer: Perhaps there should be numbered list. This step looks like it can be glided over and result in misconfiguration

```suggestion
2. Update the `sqlnet.ora`, `listener.ora`, and `tnsnames.ora` to allow TCPS connections on your Oracle Database. 
```

The configuration block above should be `1.` with the config example indented

    only_custom_queries: true
```

> #### Connecting to Oracle via TCPS
> 
> 
> 1. To connect to Oracle via TCPS (TCP with SSL), uncomment the `protocol` configuration option and select `TCPS`.
> Update the `server` option to set the TCPS server to monitor.

Reviewer 2: ```suggestion
#### Connect to Oracle through TCPS

1. To connect to Oracle through TCPS (TCP with SSL), uncomment the `protocol` configuration option and select `TCPS`. Update the `server` option to set the TCPS server to monitor.
```

> ```yaml

Reviewer: needs indentation

    
        ## @param service_name - string - required
        ## The Oracle Database service name. To view the services available on your server,
        ## run the following query:
        ## `SELECT value FROM v$parameter WHERE name='service_names'`
        #
        service_name: "<SERVICE_NAME>"
    
        ## @param user - string - required
        ## The username for the user account.
        #
        user: <USER>
    
        ## @param password - string - required
        ## The password for the user account.
        #
        password: "<PASSWORD>"
    
        ## @param protocol - string - optional - default: TCP
        ## The protocol to connect to the Oracle Database Server. Valid protocols include TCP and TCPS.
        ##
        ## When connecting to Oracle Database via JDBC, `jdbc_truststore` and `jdbc_truststore_type` are required.
        ## More information can be found from Oracle Database's whitepaper:
        ##
        ## https://www.oracle.com/technetwork/topics/wp-oracle-jdbc-thin-ssl-130128.pdf
        #
        protocol: TCPS
    ```

2. Update the `sqlnet.ora`, `listener.ora`, and `tnsnames.ora` to allow TCPS connections on your Oracle Database. 

> ##### TCPS via Oracle Instant Client

Reviewer 2: ```suggestion
##### TCPS through the Oracle Instant Client
```

> If you are connecting to Oracle Database using the Oracle Instant Client, verify that the Datadog Agent will be able to connect to your database by using the `sqlplus` command line tool using the information inputted in your configuration options:

Reviewer 2: ```suggestion
If you are connecting to Oracle Database using the Oracle Instant Client, verify that the Datadog Agent is able to connect to your database. Use the `sqlplus` command line tool with the information inputted in your configuration options:
```

```shell
sqlplus <USER>/<PASSWORD>@(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCPS)(HOST=<HOST>)(PORT=<PORT>))(SERVICE_NAME=<SERVICE_NAME>)))
```

> ##### TCPS via JDBC

Reviewer 2: ```suggestion
##### TCPS through JDBC
```

> If you are connecting to Oracle Database using JDBC, you will need to also specify `jdbc_truststore_path`, `jdbc_truststore_type`, and optionally `jdbc_truststore_password` if there is a password on the truststore. 

Reviewer 2: ```suggestion
If you are connecting to Oracle Database using JDBC, you also need to specify `jdbc_truststore_path`, `jdbc_truststore_type`, and `jdbc_truststore_password` (optional) if there is a password on the truststore. 
```

Note: `SSO` truststores don't require passwords.

```yaml
    # In the `instances:` section
    ...

    ## @param jdbc_truststore_path - string - optional
    ## The JDBC truststore file path.
    #
    jdbc_truststore_path: /path/to/truststore

    ## @param jdbc_truststore_type - string - optional
    ## The JDBC truststore file type. Supported truststore types include JKS, SSO, and PKCS12.
    #
    jdbc_truststore_type: SSO

    ## @param jdbc_truststore_password - string - optional
    ## The password for the truststore when connecting via JDBC.
    #
    # jdbc_truststore_password: <JDBC_TRUSTSTORE_PASSWORD>
```

> More information regarding connecting to Oracle Database via TCPS on JDBC can be found in the official [Oracle whitepaper][17].

Reviewer 2: ```suggestion
For more information about connecting to the Oracle Database through TCPS on JDBC, see the official [Oracle whitepaper][17].
```
